### PR TITLE
Replay Next Reconciliation: Consolidate many files into variables.css

### PR DESCRIPTION
**Pull request by jonbell-lot23**
This ticket is all about moving more things into variables.css rather than having stuff scattered into lots of places.

### CODE REVIEW

**Comment by jonbell-lot23** [comment]
Seeing some failures so I'm removing Brian as reviewer for now

**Comment by jasonLaster** [comment]
Nice to see this check 
Screenshot 2023-01-05 at 9 21 04 PM

**Comment by jonbell-lot23** [comment]
> Nice to see this check
> Screenshot 2023-01-05 at 9 21 04 PM

Holy cow -- just like how it's hard to put a price tag on Replay solving an impossible bug, that's how I feel about Delta here. Before I could *assume* I did my work correctly, but seeing "0 of 360 snapshots are different" feels amazing

**Comment by jonbell-lot23** [comment]
Delta tells me that the screens didn't change, which is good: https://delta.replay.io/project/dcb5df26?branch=2023-01-05-designsystem-4

Once I get this reviewed + squashed and merged, the finish line for this stage is near. I'll move the rest of the old variables.css into the new one, and at that point at least we'll only have one CSS file powering our design system.

File: packages/replay-next/pages/variables.css [hunk 1]
```diff
@@ -2,6 +2,108 @@
   --font-family-default: Inter, sans-serif;
   --font-family-monospace: Menlo, ui-monospace, SFMono-Regular, Monaco, Consolas, Liberation Mono,
     Courier New, monospace;
+
+  /* Migration note: everyone below this was removed from src/devtools/client/themes/variables.css */
```
Comment (bvaughn) [inline packages/replay-next/pages/variables.css:6]: These variables look like they're in the wrong place. They're mostly _colors_ but they're outside of a light/dark theme block.

Colors should all be within either the `:root.theme-dark` or `:root.theme-light` block.

File: packages/replay-next/pages/variables.css [hunk 2]
```diff
@@ -532,6 +725,98 @@
...
+  --table-splitter-color: rgba(0, 0, 0, 0.15);
+  --table-zebra-background: rgba(0, 0, 0, 0.05);
+  --sidemenu-selected-arrow: url("/images/item-arrow-ltr.svg");
+  --sidemenu-selected-arrow-rtl: url("/images/item-arrow-rtl.svg");
+
+  /* src/devtools/client/themes/perf.css */
+  --highlight-color: var(--blue-55);
+  --slider-thumb-color: var(--grey-50);
+  --slider-track-color: var(--grey-30);
+
```
Comment (bvaughn) [inline packages/replay-next/pages/variables.css:819]: I diffed the above light and dark theme blocks.

Light theme has the following variables that dark theme doesn't:
```
--tab-standard-color
--tab-standard-hover-bgcolor
--tab-bgcolor
```

Dark theme has the following variables that light theme doens't:
```
--cm-variable
--cm-property
--cm-atom
--jellyfish-bgcolor
--timejump-text
--theme-console-input-bgcolor
--theme-toggle-handle-bgcolor
--theme-toggle-handle-bgcolor-hover
--theme-toggle-bgcolor
--theme-toggle-color
--toolbarbutton-focus-bgcolor
--toolbox-bgcolor
--unloaded-region-img
--title-hover-bgcolor
--theme-toolbar-hover
```

There shouldn't be any differences in variables defined between the themes. It either indicates that the variables aren't being used, or one of the themes is broken.

File: src/devtools/client/themes/boxmodel.css [hunk 3]
```diff
@@ -1,27 +1,3 @@
...
-:root.theme-dark {
-  --marginbox-color: #73764a;
-  --borderbox-color: var(--grey-70);
-  --paddingbox-color: #6657a6;
-  --contentbox-color: #407aa4;
-  --marginbox-border-color: #bdca00;
-  --contentbox-border-color: #00b8ff;
-  --position-border-color: var(--grey-40);
-}
-
```
Comment (bvaughn) [inline src/devtools/client/themes/boxmodel.css:24]: Glad to see these blocks being removed

File: packages/replay-next/pages/variables.css [hunk 4]
```diff
@@ -2,6 +2,108 @@
...
+  --grey-43: #a4a4a4;
+  --grey-45: #939395;
+  --grey-50: #737373;
+  --grey-55: #5c5c5f;
+  --grey-60: #4a4a4f;
+  --grey-70: #38383d;
+  --grey-80: #2a2a2e;
+  --grey-90: #0c0c0d;
+  --grey-90-a10: rgba(12, 12, 13, 0.1);
+  --grey-90-a30: rgba(12, 12, 13, 0.3);
```
[the remark this answers is not in the document]
Comment (jonbell-lot23) [reply packages/replay-next/pages/variables.css:106]: Correct. This happens throughout our code. Mostly from Firefox variables we got for free, but it's common in design systems. Descriptive names like "button-color" change per theme. But names that directly refer to color like "gray-10" (also black and white) are indeed theme agonistic.

File: packages/replay-next/pages/variables.css (hunk 2 above)
Comment (jonbell-lot23) [reply packages/replay-next/pages/variables.css:819]: Thank you! Yes, these are variables that over time have fallen out of sync.

My plan is rooting those out once we land these big pieces. This PR was more of a direct copy/paste, then I'll loop back around and address this over here: https://linear.app/replay/issue/DES-218/track-down-variables-that-arent-being-used-across-themes

**Comment by bvaughn** [comment]
FYI #8505 is going to generate some updated fixture data and may impact this branch. I'll try to push this one over the finish line if it does (and if you want help).

**Comment by bvaughn** [comment]
I tried to rebase this, and to merge main into it. I even tried squashing all of the commits down and rebasing that (https://github.com/replayio/devtools/compare/jon-1?expand=1) but there are conflicts that I don't know to resolve. Afraid I'm going to need your help on those, Jon 😞 Specifically I don't know how to resolve the conflicts in `packages/replay-next/pages/variables.css`

**Comment by jonbell-lot23** [comment]
> I tried to rebase this, and to merge main into it. I even tried squashing all of the commits down and rebasing that (https://github.com/replayio/devtools/compare/jon-1?expand=1 and https://github.com/replayio/devtools/compare/jon-2?expand=1) but there are conflicts that I don't know to resolve. Afraid I'm going to need your help on those, Jon 😞 Specifically I don't know how to resolve the conflicts in `packages/replay-next/pages/variables.css`

Thanks Brian. I did some merges over here: https://github.com/replayio/devtools/pull/8508

I am currently checking by hand and waiting on Delta to see how it's looking.

**Comment by bvaughn** [comment]
Cool. I'll ignore this PR in favor of #8508 then. (Should we close this one and #8497?)

**Comment by jonbell-lot23** [comment]
> Cool. I'll ignore this PR in favor of #8508 then. (Should we close this one and #8497?)

Yup, that was my thinking -- glad I wasn't off base with that assumption.

**Pull request closed without merging**